helm/examples/azure/sa/create-sa.sh

```diff
@@ -0,0 +1,80 @@
+#!/bin/bash
+WORK_DIR=$(dirname $0)
+source ${WORK_DIR}/env-sa
+
+if [[ -z "$SA_RESOURCE_GROUP" ]] || [[ -z "$SA_LOCATION" ]] || [[ -z "$STORAGE_ACCOUNT_NAME" ]]
+then
+   echo "Miss one of SA_RESOURCE_GROUP, SA_LOCATION and STORAGE_ACCOUNT_NAME"
+   exit 1
+fi
+
+[[ -n "$SUBSCRIPTION" ]] && az account set --subscription $SUBSCRIPTION
+
+VALUES_FILE=${WORK_DIR}/../hpcc-azurefile/values.yaml
+SECRET_NAME_INPUT=$(cat $VALUES_FILE | grep "^[[:space:]]*secretName:"|cut -d':' -f2|sed 's/[[:space:]]//g')
+SECRET_NAMESPACE_INPUT=$(cat $VALUES_FILE | grep "^[[:space:]]*secretNamespace:"|cut -d':' -f2|sed 's/[[:space:]]//g')
+SHARE_NAME_INPUT=$(cat $VALUES_FILE | grep "^[[:space:]]*shareName:"|cut -d':' -f2|sed 's/[[:space:]]//g')
+[[ -n "$SECRET_NAME_INPUT" ]] && SECRET_NAME=${SECRET_NAME_INPUT}
+[[ -n "$SECRET_NAMESPACE_INPUT" ]] && SECRET_NAMESPACE=${SECRET_NAMESPACE_INPUT}
+[[ -n "$SHARE_NAME_INPUT" ]] && SHARE_NAME=${SHARE_NAME_INPUT}
+
+if [[ -z "$SECRET_NAME" ]] || [[ -z "$SECRET_NAMESPACE" ]] || [[ -z "$SHARE_NAME" ]]
+then
+   echo "Miss one of SECRET_NAME, SECRET_NAMESPACE and SHARE_NAME"
+   exit 1
+fi
+
+# Create a resource group
+# Check Resource Group
+rc=$(az group exists --name ${SA_RESOURCE_GROUP})
+if [ "$rc" != "true" ]
+then
+  az group create --name ${SA_RESOURCE_GROUP} --location ${SA_LOCATION}
+fi
+
+az storage account check-name -n $STORAGE_ACCOUNT_NAME | \
+	grep -q "\"nameAvailable\":[[:space:]]*true"
+if [ $? -ne 0 ]
+then
+  echo "Invalid storage account name: $STORAGE_ACCOUNT_NAME"
+  exit 1
+fi
+
+az storage account check-name -n $STORAGE_ACCOUNT_NAME | \
+	grep -q "\"reason\":[[:space:]]\"AlreadyExist\""
+if [ $? -ne 0 ]
+then
+  # Create a storage account
+  az storage account create \
+    -n $STORAGE_ACCOUNT_NAME \
+    -g $SA_RESOURCE_GROUP \
+    -l $SA_LOCATION \
+    --sku $SA_SKU
+
+fi
+# Export the connection string as an environment variable,
+# this is used when creating the Azure file share
+export AZURE_STORAGE_CONNECTION_STRING=$(az storage account show-connection-string \
+  -n $STORAGE_ACCOUNT_NAME -g $SA_RESOURCE_GROUP -o tsv)
+
+az storage share exists --connection-string $AZURE_STORAGE_CONNECTION_STRING \
+  --name  $SHARE_NAME | grep -q  "\"exists:\":[[:space:]]\"false\""
+if [ $? -ne 0 ]
```

Comment:
Shouldn't this be a -eq test?  So if the share does not exist then the create is called?

Reply:
fixed